### DISCUSSION TITLE
Friendlycaptcha

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uAssets#14916] [General]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

`https://friendlycaptcha.com/demo`

### Describe the issue

- Friendlycaptcha works on an unmodified browser profile with default lists/settings. 
- It does not work in my modified settings with lists (entirely disabling uBO suffices to fix the issue). 

I narrowed the issue down to a filter that I added after seeing it recommended:
```adb
*$csp=worker-src 'none'
@@||example.org^$csp=worker-src 'none'
```
Hadn't posed issues before and I would like to keep it, if possible. Any way to make this compatible with the captcha?

Thanks a lot for the help.
Best,
Candy

### uBlock Origin version

1.44.2

### Browser name and version

FF 105

### Settings
```
Hide placeholders of blocked elements
Show the number of blocked requests on the icon
Make use of context menu where appropriate
Disable pre-fetching (to prevent any connection for blocked network requests) 
Disable hyperlink auditing 
Block CSP reports 
Uncloak canonical names 
Block remote fonts 
```

**Comment by MasterKia** [comment]
> a filter that I added after seeing it recommended

Where was it recommended and why?

**Comment by mapx-** [comment]
bad idea using such a filter, just remove it (maybe it was "recommended" during that mining period)

**Comment by ghost** [comment]
Thanks for the replies. Looked around but can't find the source anymore; twas a page on privacy settings for uBO. Can't say that it's been an issue so far. 

Would you mind expanding a bit on why this is bad and should be removed? Pros and cons?

EDIT: 
`||$csp=worker-src 'none',domain=~friendlycaptcha.com`
seems to have solved the issue. Would that be a good middle ground?

**Reply by mapx-** [reply]
It will break all service worker / shared worker / captcha etc
If you have a certain issue with some service worker just report in uAssets and we'll investigate if should be blocked

Don't add all the filters you find in the wild

**Reply by ghost** [reply]
Thanks for the advice. Since we're talking about privacy, however, I feel like if I just allow all, there won't be an issue to report in the first place. Any good reason not to block if there's no major issue? (I'll keep in mind that captcha can be fixed with an exception).

**Reply by MasterKia** [reply]
For more privacy, see:
https://github.com/uBlockOrigin/uBlock-issues/wiki/Per-site-switches#no-remote-fonts

https://github.com/uBlockOrigin/uBlock-issues/wiki/Dynamic-filtering:-to-easily-reduce-privacy-exposure

https://github.com/uBlockOrigin/uBlock-issues/wiki/Dynamic-filtering:-Benefits-of-blocking-3rd-party-iframe-tags

https://github.com/uBlockOrigin/uBlock-issues/wiki/Blocking-mode:-medium-mode

Also you can block all third-party cookies.

**Reply by ghost** [reply]
Oh I'm already in hard mode with everything blocked unless approved :)
